src/Operation/BulkWrite.php

```diff
@@ -229,7 +229,7 @@ public function __construct(string $databaseName, string $collectionName, array
                         throw InvalidArgumentException::invalidType(sprintf('$operations[%d]["%s"][1]', $i, $type), $args[1], 'array or object');
                     }
 
-                    if (! is_first_key_operator($args[1]) && ! is_pipeline($args[1])) {
+                    if ($args[1] === [] || ! is_first_key_operator($args[1]) && ! is_pipeline($args[1])) {
```

Comment:
I need to check the spec to see if an empty array is accepted in the following cases:
- BulkWrite UPDATE_MANY and UPDATE_ONE
- FindOneAndUpdate
- ReplaceOne
- Update, UpdateMany, UpdateOne

Reply:
So I'll change the behavior for `ReplaceOne` that is currently rejecting empty array.

Reply:
Should be fine - just make sure it is covered by a test.

FWIW, the shell behaves the same way:
```
test> // Replace with empty replacement
test> db.test.replaceOne({_id: 1}, {})
{
  acknowledged: true,
  insertedId: null,
  matchedCount: 1,
  modifiedCount: 1,
  upsertedCount: 0
}
test> // Update with empty document
test> db.test.updateOne({_id: 1}, {})
MongoInvalidArgumentError: Update document requires atomic operators
test> // Update with empty pipeline
test> db.test.updateOne({_id: 1}, [])
MongoInvalidArgumentError: Update document requires atomic operators
```

Reply:
Thanks for checking, I'll adopt this reflex.

Test case already added: https://github.com/mongodb/mongo-php-library/pull/1078/files#diff-6e13b3e2ae70bee5981233671f8a994c0111a3cf84e2f958e723bdd08c9c83d2R45